### DISCUSSION TITLE
Converting Base64 to MP3 or WAV file

### DISCUSSION THREAD

**Discussion by Maya-Taliaferro** [jspsych/jsPsych#2772] [Q&A]
Hi everyone, 

I have an experiment which records audio from a participant's mic utilizing html-audio-response. This plugin records the audio in Base64 format. I was wondering if anyone knows any scripts (python, Matlab or R all ok) and/or programs that can read in Base64 and convert it to an mp3 or wav file? Ideally I'd like to do this for multiple Base64 strings. 

Thanks so much!

**Comment by Rots** [comment]
I ended up using the apprach:
1. programmatically decode base64 stream with R
2. save the resulting bytes as `outfile.webm`
3. convert to wav `ffmpeg -i outfile.webm outfile.wav`

also see https://github.com/jspsych/jsPsych/discussions/1202#discussioncomment-134300 and https://github.com/jspsych/jsPsych/discussions/796#discussioncomment-267645 for examples
